### PR TITLE
Lambda hooks

### PR DESCRIPTION
**Pull request by tqhoughton**
Added hooks to lambda file so that we can use these hooks in other testing frameworks besides ava.

Also added typescript definitions for the lambda module.

I can split this up into multiple PRs if needed.

### CODE REVIEW

**Comment by tqhoughton** [comment]
Thanks @Charliekenney23 for the crypto package fix. This is ready for review now.

File: src/index.d.ts [hunk 1]
```diff
@@ -19,3 +19,83 @@ declare namespace dynamodb {
   ): void;
   export function dynamoDBTestHooks(useUniqueTables?: boolean): Hooks;
 }
+
+declare namespace lambda {
+  interface Environment {
+    [key:string]: string | null;
```
Comment (Charliekenney23) [inline src/index.d.ts]: I think `Record<string, string>` would be best here.

Comment (tqhoughton) [reply src/index.d.ts]: Wouldn't that make it unable to be `null`? This can be null if we want to delete attributes from the environment

Comment (Charliekenney23) [reply src/index.d.ts]: `delete` will remove the attribute binding though. Environment variables can only be represented as strings, a key miss would result in undefined.

Comment (tqhoughton) [reply src/index.d.ts]: What are you referring to when you say `delete`? I know they can only be represented as strings, but when we pass in environments from functions it uses the `createEnvironmentVariables` function which expects keys that need to be deleted to be provided with `null` values.

Comment (Charliekenney23) [reply src/index.d.ts]: Oh, I misunderstood the use case

**Comment by tqhoughton** [comment]
@mdlavin I fixed those types. Any final concerns from anyone before I merge?

**Comment by mdlavin** [comment]
Looks good to me

**Pull request merged**